pkg/rules/engine/engine.go

```diff
@@ -23,7 +24,7 @@ type Config struct {
 
 // Engine is a rule-engine that can process events coming from a set of input sources against a set of loaded signatures, and report the signatures' findings
 type Engine struct {
-	logger          log.Logger
+	logger          logger.Logger
```

Comment:
Thinking if we should just use our global logger here as we probably would in the rest of the project, @josedonizetti

Reply:
I did not quite understand this :/

Reply:
Basically since we set our log level globally and use the global logger almost everywhere else (for example `log.Debug(...)` and not `logger.Debug(...)`), it might be more appropriate now to just remove this dependency injection of logger and just use the package's global function.
I do want a second opinion on this though so I tagged @josedonizetti.

Reply:
sorry for the delay guys! +1 for what @NDStrahilevitz  said, although I like injecting logs, tracee is moving towards the global solution, so let's be consistent

Reply:
But then if we revert back to the log level set globally, will this log in the logger itself? Won't it rather log according to the inbuilt "log" package and not in context with our own logger as we need?

PS: I do not find any use of `log.Debug()` in Tracee, I migrated the left over ones to use **our logger** itself in this PR

Reply:
@ShubhamPalriwala sorry, not sure I understood your concern. I think the idea is to be consistent about how we do logging, instead of doing dependency injection in a few places and global in others.

Reply:
@ShubhamPalriwala There is no revert since we do not intend to use the injected logger, we will use the single global logger which already has a set logging level.
You are correct we don't use `log.Debug()` currently, but just like what your PR is starting, we need to refactor the rest of our project to use it.

Reply:
Alright! So for now, we should also remove this logger dependency injection from the `engine/engine.go` right?

Reply:
Umm do you mean the "log" package? Basically keeping how things were before this?

Reply:
I meant replace this https://github.com/aquasecurity/tracee/pull/2224/files/3fc1ef9112422157ceb822b6a7f307387952d274#diff-17c4ab74a504419b8d75f98b44060023e0b680ad2894f265398ae9af42068b3bL65 whit:

```go

logger.Error("loading signature: + err.Error())
```

Getting rid of the Engine injected one.

Reply:
Made and pushed the changes @geyslan @josedonizetti @NDStrahilevitz, Please take a look whenever you get time!